src/metabolic_ninja/resources.py

```diff
@@ -185,6 +186,164 @@ def get(self):
         return PRODUCT_LIST
 
 
+class JobRowResource(MethodResource):
+    def get(self, job_id, row_id):
+        job_id = int(job_id)
+        try:
+            job = (
+                DesignJob.query.filter(DesignJob.id == job_id)
+                .filter(
+                    DesignJob.project_id.in_(g.jwt_claims["prj"])
+                    | DesignJob.project_id.is_(None)
+                )
+                .one()
+            )
+        except NoResultFound:
+            return (
+                {"error": f"Cannot find any design job with id {job_id}."},
+                404,
+            )
+        else:
+            if job.is_complete():
+                status = 200
+            else:
+                status = 202
+            pathway = next(
+                (
+                    x
+                    for x in job.result["diff_fva"]
+                    + job.result["opt_gene"]
+                    + job.result["cofactor_swap"]
+                    if x["id"] == row_id
+                ),
+                None,
+            )
+            if not pathway:
+                return (
+                    {"error": f"Cannot find any row with id {row_id} in job {job_id}."},
+                    404,
+                )
+            result = get_tabular_data(pathway)
+            response = make_response(result, status)
+            response.headers["Content-Type"] = "text/csv"
+            response.headers["Content-Disposition"] = "attachment"
+            return response
+
+
+def get_tabular_data(pathway):
```

Comment:
This logic isn't specific to the HTTP resources so it doesn't belong here. Same with all the added functions below. You could inline it if it's a one-off thing, but it looks to me like it could make sense to encapsulate this in the data model, for example as methods on the `Job` ORM class.